Sources/SwiftProtobuf/FieldNameMap.swift

```diff
@@ -8,6 +8,82 @@
 //
 // -----------------------------------------------------------------------------
 
+/// Proto field names are always ASCII, so we can improve the
+/// performance of some of our parsing by storing them as
+/// arrays of UInt8 to avoid the overhead of creating and
+/// comparing Unicode Strings.
```

Comment:
> A side effect of this is that ASCIIString will have to be public since it will become the FieldNameMaps.Names payload type instead of String.

Does it?  Would it be a problem if we made the payload type be StaticString and kept the custom type private?